### PR TITLE
[HttpFoundation] Make MimeTypeExtensionGuesser case insensitive

### PR DESCRIPTION
**Pull request by vermeirentony**
| Q             | A
| ------------- | ---
| Branch?       | 3.4 
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no  
| Deprecations? | no
| Tests pass?   | yes 
| Fixed tickets | #...   
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Some mime types have a camelCase word in them.
The Apache HTTPD project list items are all lower case.
So I suggest making the $mimeType string lowercase while checking the array key.
That way, we can keep the list in sync.

Example: xlsm file mime type is `application/vnd.ms-excel.sheet.macroEnabled.12`
The key that matches the xlsm extension in the `$defaultExtensions` array is `application/vnd.ms-excel.sheet.macroenabled.12`

Example xlsm file:
https://github.com/vermeirentony/xlsm-example

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/File/MimeType/MimeTypeExtensionGuesser.php [hunk 1]
```diff
@@ -808,6 +808,6 @@ class MimeTypeExtensionGuesser implements ExtensionGuesserInterface
      */
     public function guess($mimeType)
     {
-        return isset($this->defaultExtensions[$mimeType]) ? $this->defaultExtensions[$mimeType] : null;
+        return isset($this->defaultExtensions[strtolower($mimeType)]) ? $this->defaultExtensions[strtolower($mimeType)] : null;
```
Comment (ro0NL) [inline src/Symfony/Component/HttpFoundation/File/MimeType/MimeTypeExtensionGuesser.php]: perhaps store a lowercased `$lcMimeType` once, and maybe try isset() on the original mime type first (could be a legit optimization).

**Review by ro0NL — APPROVED** [review]
https://stackoverflow.com/questions/12869278/are-mimetypes-lowercase-case-insensitive

**Comment by ro0NL** [comment]
this needs a patch for master as well, where `MimeTypes` is a new class.

File: src/Symfony/Component/HttpFoundation/File/MimeType/MimeTypeExtensionGuesser.php (hunk 1 above)
Comment (vermeirentony) [reply src/Symfony/Component/HttpFoundation/File/MimeType/MimeTypeExtensionGuesser.php]: @ro0NL I made the suggested optimizations, can you review them?

Comment (ro0NL) [reply src/Symfony/Component/HttpFoundation/File/MimeType/MimeTypeExtensionGuesser.php]: i think we should add a test also :)

**Comment by nicolas-grekas** [comment]
@vermeirentony can you add a test case please?

**Comment by vermeirentony** [comment]
@nicolas-grekas 
test case (3.4) can be found on https://github.com/vermeirentony/mimetype-testcase

**Comment by Simperfit** [comment]
@vermeirentony Thanks for the contribution,

I think by testcase Nicolas was meaning a test directly in the framework with PHPUnit, could you do that please ?

**Comment by vermeirentony** [comment]
@Simperfit sure, will write one later today

**Comment by vermeirentony** [comment]
@nicolas-grekas I just added the unit test

**Review by ro0NL — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
@vermeirentony Can you squash your commits before merging? Right now, you've used 2 different identities for your commits. You probably need to keep the one here on Github.

**Comment by vermeirentony** [comment]
@fabpot squashed the 3 commits into one

**Review by Simperfit — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @vermeirentony

**Pull request merged**